### PR TITLE
Fixes #4704

### LINKED ISSUE #4704: [Bug] [delete browsing data] disable checkboxes or hide items when there is no data to be cleared
#### Expected behavior
When I see several checkboxes I want to mark them all to delete everything, but if there is no content to be deleted, the checkbox remains the same. IMO these elements should have a more washes out color, indicating they are not available. 

### Actual behavior
Screenshot_20190812-225414_Firefox Preview

### Device information

* Android device: Galaxy A5 2017 8.0 
* Fenix version: 190813

### PR DESCRIPTION
**Pull request by shahsurajk**
### Pull Request checklist
- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/fenix/blob/master/CHANGELOG.md) or does not need one
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

Fixes #4704 

### How:
* The logic for enabling and disabling the views is present already, this commit overrides the function in the respective view and disables the checkbox and uses a pre-used alpha value (the one used by the delete button in the fragment consuming it as a standard)

The PR has no test cases added as no change in logic was done.

### CODE REVIEW

File: app/src/main/java/org/mozilla/fenix/settings/DeleteBrowsingDataItem.kt [hunk 1]
```diff
@@ -61,4 +66,10 @@ class DeleteBrowsingDataItem @JvmOverloads constructor(
             subtitle.text = resources.getString(subtitleId)
         }
     }
+
+    override fun setEnabled(enabled: Boolean) {
+        super.setEnabled(enabled)
+        alpha = if (enabled) ENABLED_ALPHA else DISABLED_ALPHA
+        checkbox.isEnabled = enabled
```
Comment (NotWoods) [inline app/src/main/java/org/mozilla/fenix/settings/DeleteBrowsingDataItem.kt]: Is this line needed? As far as I can tell, I can't use the checkbox if `DeleteBrowsingDataItem` is disabled.

File: CHANGELOG.md [hunk 2]
```diff
@@ -6,6 +6,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 ### Added
+- #4704 - Adds ability to show a different alpha and disables checkbox
```
Comment (NotWoods) [inline CHANGELOG.md]: It's not clear where this change affects the app. Perhaps something like "Delete browsing data items will show a different alpha when there is nothing to delete" would be more clear.

Comment (shahsurajk) [reply CHANGELOG.md]: Agreed, there's a bit of ambiguity here, let me refactor this and push again.

File: app/src/main/java/org/mozilla/fenix/settings/DeleteBrowsingDataItem.kt (hunk 1 above)
Comment (shahsurajk) [reply app/src/main/java/org/mozilla/fenix/settings/DeleteBrowsingDataItem.kt]: Yeah, got this wrong, re-pushing.

Comment (shahsurajk) [reply app/src/main/java/org/mozilla/fenix/settings/DeleteBrowsingDataItem.kt]: Pushed change to include this.

File: CHANGELOG.md (hunk 2 above)
Comment (shahsurajk) [reply CHANGELOG.md]: Pushed change to include this.

**Pull request merged**